File: src/main/java/com/joseph/spareapi/config/MongoConfiguration.java
package com.joseph.spareapi.config;

import com.mongodb.MongoClient;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.data.mongodb.config.AbstractMongoConfiguration;
import org.springframework.data.mongodb.gridfs.GridFsTemplate;

@Configuration
public class MongoConfiguration extends AbstractMongoConfiguration {
    @Value("${spring.data.mongodb.database}")
    private String databaseName;
    @Value("${spring.data.mongodb.uri}")
    private String mongoAddress;
    @Override @Bean
    public MongoClient mongoClient() {
        return new MongoClient(mongoAddress);
    }

    @Override
    protected String getDatabaseName() {
        return databaseName;
    }
    @Bean
    public GridFsTemplate gridFsTemplate() throws Exception {
        return new GridFsTemplate(mongoDbFactory(),mappingMongoConverter());
    }
}

File: src/main/scala/com/joseph/config/MongoConfiguration.scala
package com.joseph.config

import com.mongodb.MongoClient
import org.springframework.beans.factory.annotation.Value
import org.springframework.context.annotation.{Bean, Configuration}
import org.springframework.data.mongodb.config.AbstractMongoConfiguration
import org.springframework.data.mongodb.gridfs.GridFsTemplate

@Configuration
class MongoConfiguration extends AbstractMongoConfiguration {
  private val databaseName: String = "sparedb"
  @Value("${spring.data.mongodb.uri}")
  private var mongoAddress: String = _

  @Bean
  override def mongoClient(): MongoClient = {
    new MongoClient(mongoAddress)
  }

  override def getDatabaseName: String = {
    databaseName
  }

  @Bean
  @throws[Exception]
  def gridFsTemplate = new GridFsTemplate(mongoDbFactory, mappingMongoConverter)
}
